src/test/test-http.c

```diff
@@ -0,0 +1,183 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Intel Corporation nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "test.h"
+#include "sol-util.h"
+#include "sol-http.h"
+#include "sol-vector.h"
+
+DEFINE_TEST(test_split_invalid_urls);
+
+static void
+test_split_invalid_urls(void)
+{
+    struct sol_str_slice protocol, server, fragment, path;
+    struct sol_http_param params;
+    int r, port;
+    unsigned int i;
+    const char *invalid_urls[] = {
+        "",
+        "!231312312#$http$//www.intel.com",
+        "http//www.intel.com",
+        "      http   :    //www.intel.com",
+        "http:www.intel.com"
+        "http://www.intel.com:asd!%%%!!23332182/",
+        "http://",
+        "www.intel.com",
+        "http://\"",
+        "http://www.intel.com:80:80",
+    };
```

Comment:
you better use the URLs I pointed out with a more extensive test case. URI is important and easy to get wrong, so we better do it right from the beginning :-)

while at that, would be nice to see gcov results to see if we're exploring all uri split in our code.